drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/report.rb

```diff
@@ -380,6 +380,154 @@ def average_days_before_entry
+      goal = [goal, goal, goal]
+      {
+        labels: labels,
+        data: {
+          'Goal' => goal,
+          'Average' => [0, days, 0],
+        },
+      }
+    end
+
+    def timeliness_entry_goal
+      goal_config.entry_date_entered_length # days
+    end
+
+    def timeliness_exit_goal
+      goal_config.exit_date_entered_length # days
+    end
+
+    def time_in_enrollment_chart
+      data = {}
+      [
+        :es,
+        :so,
+        :ph,
+      ].each do |project_type_slug|
+        next unless any_enrollments_in_type?(project_type_slug)
+
+        data[project_type_slug.upcase] = [
+          enrollments_of_length(0..30, project_type_slug),
+          enrollments_of_length(31..180, project_type_slug),
+          enrollments_of_length(181..364, project_type_slug),
+          enrollments_of_length(365.., project_type_slug),
+        ]
+      end
+      {
+        labels: ['1 month or less', '1 to 6 months', '6 to 12 months', '12 months or greater'],
+        data: data,
+        ranges: {
+          '1 month or less' => '0..30',
+          '1 to 6 months' => '31..180',
+          '6 to 12 months' => '181..364',
+          '12 months or greater' => '365..Infinity',
+        },
+      }
+    end
+
+    def any_enrollments_in_type?(project_type_slug)
+      project_types = HudUtility2024.residential_project_type_numbers_by_code[project_type_slug]
+      enrollments.where(project_type: project_types).exists?
+    end
+
+    def average_time_in_project_type(project_type_slug)
+      project_types = HudUtility2024.residential_project_type_numbers_by_code[project_type_slug]
+      scope = enrollments.where(project_type: project_types)
+      scope = scope.where(move_in_date: nil) if project_type_slug == :ph
+      count = scope.count
+      sum = scope.sum(:lot)
+      return 0 if count.zero?
+
+      sum / count
+    end
+
+    def percent_enrollments_over_one_year(project_type_slug)
+      numerator = enrollments_of_length(365.., project_type_slug)
+      project_types = HudUtility2024.residential_project_type_numbers_by_code[project_type_slug]
+      scope = enrollments.where(project_type: project_types)
+      scope = scope.where(move_in_date: nil) if project_type_slug == :ph
+      denominator = scope.count
+      return 0 if denominator.zero?
+
+      ((numerator / denominator.to_f) * 100).round
+    end
+
+    private def enrollments_of_length(range, project_type_slug)
+      project_types = HudUtility2024.residential_project_type_numbers_by_code[project_type_slug]
+      scope = enrollments.where(lot: range, project_type: project_types)
+      scope = scope.where(move_in_date: nil) if project_type_slug == :ph
+      scope.count
+    end
+
+    def goal_segments
+      @goal_segments ||= goal_config&.active_segments
+    end
+
+    def completeness_target
+      return nil unless goal_segments.present?
+
+      # Return the bottom of the top segment
+      goal_segments.values.map(&:first).max.inspect
```

Comment:
usually `inspect` is for debugging. Maybe should be `to_s`?

Reply:
Huh, that was just some left-over debugging I guess.  Thank you!